chain/action_test_helpers.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+type ActionTest struct {
+	Action Action
+
+	Rules     Rules
+	State     state.Mutable
+	Timestamp int64
+	Actor     codec.Address
+	ActionID  ids.ID
+
+	ExpectedOutputs   [][]byte
+	ExpectedErr       error
+	ContainsErrString string
+}
+
+type ActionTestSuite struct {
+	Tests map[string]ActionTest
+
+	Setup    func()
+	Teardown func()
```

Comment:
I'm actually not sure if we will need those, I can remove them for this PR and we can add them back later if so.

Reply:
Removed for now. These hook functions could be useful if we pass a reference to the storage in order to be able to assert stuff on the written storage.